packages/wdio-browserstack-service/src/util.ts

```diff
@@ -13,14 +15,22 @@ import type { GitRepoInfo } from 'git-repo-info'
 import gitRepoInfo from 'git-repo-info'
 import gitconfig from 'gitconfiglocal'
 
-import type { UserConfig, UploadType, LaunchResponse, BrowserstackConfig } from './types.js'
+import type { UserConfig, UploadType, LaunchResponse, BrowserstackConfig, UserConfigforReporting, CredentialsForCrashReportUpload } from './types.js'
 import type { ITestCaseHookParameter } from './cucumber-types.js'
 import { BROWSER_DESCRIPTION, DATA_ENDPOINT, DATA_EVENT_ENDPOINT, DATA_SCREENSHOT_ENDPOINT } from './constants.js'
 import RequestQueueHandler from './request-handler.js'
 
+const require = createRequire(import.meta.url)
+const { version: bstackServiceVersion } = require('../package.json')
+
 const pGitconfig = promisify(gitconfig)
 const log = logger('@wdio/browserstack-service')
 
+/* User test config for build run minus PII */
+let userConfigForReporting: UserConfigforReporting = {}
+/* User credentials used for reporting crashes in browserstack service */
+let credentialsForCrashReportUpload: CredentialsForCrashReportUpload = {}
```

Comment:
I am not sure how I feel about creating root variables to store this data. It's a code smell. This should be handled in a class, e.g. a class to handle crash reporting

Reply:
removed, and moved to a separate class `crash-reporter`